ecosystem/sep-0010.md

```diff
@@ -250,6 +257,25 @@ A server may choose to issue JWTs for less than all thresholds and based on any
 
 A server that needs to support validating accounts that do not exist can require a signature of the master key of the account address for accounts that do not exist.
 
+### Verifying Client Application Identity
+
+A server may want to attribute each HTTP request to a specific client application. For example, a server and client could make a deal to elimate fees for the users of that client. SEP-10 offers an optional variation to the flow outlined in the [Abstract](#abstract) to enable this use case. 
+
+In order to use this optional feature, the organization that owns the client application must host a [SEP-1 stellar.toml](sep-0001.md) file containing a `SIGNING_KEY` attribute. This allows the SEP-10 server to verify that the challenge sent by a client is signed by the `SIGNING_KEY`, proving that the organization takes ownership of the client application. The SEP-10 server can now attribute requests made with the resulting JWT to the organization that signed the challenge.
+
+1. The client makes the initial `GET <WEB_AUTH_ENDPOINT>` request with an added `?client_domain=example.com` parameter and value
+1. The SEP-10 server generates the challenge transaction with an additional Manage Data operation containing the `<client_domain> attribution>` passed 
+1. The client obtains a signature generated using its organization's `SIGNING_KEY` and adds it to the challenge transaction. How this happens is up to the client. However:
+  1. It is highly recommended that the client-side application does not hold the secret key for the `SIGNING_KEY`
+  1. Instead, an internal service should accept a request from the organization's client application, verify the challenge sent, and generate the signature if valid
+1. The client verifies the signature and adds the it to the challenge transaction if valid
```

Comment:
I think you can drop this step, this is internal behavior of the client and there's nothing particularly dangerous about not verifying it here.
```suggestion
```